src/js/plugins/plugin.url.js

```diff
@@ -137,10 +137,12 @@ BookReader.prototype.urlUpdateFragment = function() {
     return validParams
   }, {});
 
-  const newFragment = this.fragmentFromParams(params);
+  const newFragment = this.fragmentFromParams(params, urlMode);
   const currFragment = this.urlReadFragment();
+  const currQueryString = window.location.search;
+  const newQueryString = this.queryStringFromParams(params, currQueryString, urlMode);
 
-  if (currFragment === newFragment) {
+  if (currFragment === newFragment && currQueryString === newQueryString) {
```

Comment:
QUESTION: This is the critical comparison to determine whether the URL should be rewritten. How dependent is it on parameter order? Is it smart enough to still do a correct match if it's the same parameters but in a different order? If not, should it be?

And: Should there be unit tests to check this?

Reply:
* dependent is it on parameter order? yes
* smart enough to match if different order? no
* should it be? no, according to spec: https://url.spec.whatwg.org/#interface-urlsearchparams, internal 'List' is ordered, and original query string used
* should we check for out-of-spec implementations? no

Reviewing the code I can see a few places to add units tests, they'd test implicitly for order.

Reply:
tests added